Apps/TicketManagement/ErrorHandling/WaitingQueueException.cs

```diff
@@ -0,0 +1,114 @@
+        /// <param name="memberName">The type of the method or property where the exception was generated.</param>
+        public WaitingQueueException(string detail, HttpStatusCode statusCode, string typeName, [CallerMemberName] string memberName = "")
+        {
+            this.ProblemType = "waiting-queue-exception";
+            this.Detail = detail;
+            this.Title = "Custom Exception Handling";
+            this.AdditionalInfo = "Please try again at a later time.";
+            this.Instance = $"{typeName}.{memberName}";
+            this.StatusCode = (int)statusCode;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WaitingQueueException"/> class.
+        /// </summary>
+        /// <param name="message">The message associated with the exception.</param>
+        /// <param name="innerException">The inner exception associated with the exception.</param>
+        public WaitingQueueException(string message, Exception innerException)
+            : base(message, innerException)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WaitingQueueException"/> class.
+        /// </summary>
+        /// <param name="message">The message associated with exception.</param>
+        public WaitingQueueException(string message)
+            : base(message)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WaitingQueueException"/> class.
+        /// </summary>
+        public WaitingQueueException()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WaitingQueueException"/> class.
+        /// </summary>
+        /// <param name="serializationInfo">The serialization info associated with the exception.</param>
+        /// <param name="streamingContext">The streaming context associated with the exception.</param>
+        protected WaitingQueueException(SerializationInfo serializationInfo, StreamingContext streamingContext)
+            : base(
+                serializationInfo,
+                streamingContext)
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets additional info.
+        /// </summary>
+        public string? AdditionalInfo { get; set; }
+
+        /// <summary>
+        /// Gets or sets problem type.
+        /// </summary>
+        public string? ProblemType { get; set; }
+
+        /// <summary>
+        /// Gets or sets detail.
+        /// </summary>
+        public string? Detail { get; set; }
+
+        /// <summary>
+        /// Gets or sets title.
+        /// </summary>
+        public string? Title { get; set; }
+
+        /// <summary>
+        /// Gets or sets instance.
+        /// </summary>
+        public string? Instance { get; set; }
+
+        /// <summary>
+        /// Gets or sets status code.
+        /// </summary>
+        public int StatusCode { get; set; }
```

Comment:
Can this be HttpStatusCode instead of int?

Reply:
Converted StatusCode to HttpStatusCode and moved cast to ProblemDetailConfiguration when interacting between WaitingQueueException and WaitingQueueProblemDetails.